core/src/test/java/com/graphhopper/routing/weighting/CustomWeightingTest.java

```diff
@@ -41,97 +40,142 @@
 
 import static com.graphhopper.routing.profiles.RoadClass.*;
 import static com.graphhopper.routing.weighting.TurnCostProvider.NO_TURN_COST_PROVIDER;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class CustomWeightingTest {
 
     GraphHopperStorage graphHopperStorage;
     DecimalEncodedValue avSpeedEnc;
-    BooleanEncodedValue accessEnc;
     EnumEncodedValue<RoadClass> roadClassEnc;
     EncodingManager encodingManager;
     FlagEncoder carFE;
 
-    @Before
+    @BeforeEach
     public void setUp() {
-        carFE = new CarFlagEncoder();
+        carFE = new CarFlagEncoder().setSpeedTwoDirections(true);
         encodingManager = new EncodingManager.Builder().add(carFE).build();
         avSpeedEnc = carFE.getAverageSpeedEnc();
-        accessEnc = carFE.getAccessEnc();
         roadClassEnc = encodingManager.getEnumEncodedValue(KEY, RoadClass.class);
         graphHopperStorage = new GraphBuilder(encodingManager).create();
     }
 
+    @Test
+    public void speedOnly() {
+        // 50km/h -> 72s per km, 100km/h -> 36s per km
+        EdgeIteratorState edge = graphHopperStorage.edge(0, 1, 1000, true)
+                .set(avSpeedEnc, 50).setReverse(avSpeedEnc, 100);
+        assertEquals(72, createWeighting(new CustomModel().setDistanceInfluence(0)).calcEdgeWeight(edge, false), 1.e-6);
+        assertEquals(36, createWeighting(new CustomModel().setDistanceInfluence(0)).calcEdgeWeight(edge, true), 1.e-6);
+    }
+
+    @Test
+    public void withPriority() {
```

Comment:
This test shows how the weight changes when changing the priority for a few edges with different speeds